pixeltable/functions/__init__.py

```diff
@@ -102,31 +102,34 @@ def value(self) -> float:
 
 class VideoAggregator:
     def __init__(self):
-        self.video_writer = None
-        self.size = None
+        '''follows https://pyav.org/docs/develop/cookbook/numpy.html#generating-video'''
+        self.container = None
+        self.stream = None
+        self.fps = 24
 
     @classmethod
     def make_aggregator(cls) -> 'VideoAggregator':
         return cls()
 
     def update(self, frame: PIL.Image.Image) -> None:
-        if self.video_writer is None:
-            self.size = (frame.width, frame.height)
+        if self.container is None:
             self.out_file = Path(os.getcwd()) / f'{Path(tempfile.mktemp()).name}.mp4'
-            self.tmp_file = Path(os.getcwd()) / f'{Path(tempfile.mktemp()).name}.mp4'
-            # our target codec is H.264, but it's tainted by GPL and cv2 doesn't include it, so we use MP4V instead
-            self.video_writer = cv2.VideoWriter(str(self.tmp_file), cv2.VideoWriter_fourcc(*'mp4v'), 25, self.size)
-
-        frame_array = np.array(frame)
-        frame_array = cv2.cvtColor(frame_array, cv2.COLOR_RGB2BGR)
-        self.video_writer.write(frame_array)
-
+            self.container = av.open(str(self.out_file), mode="w")
+            self.stream = self.container.add_stream("mpeg4", rate=self.fps)
+            self.stream.width = frame.width
+            self.stream.height = frame.height
```

Comment:
do you need stream.pix_fmt = "yuv420p" anywhere?

Reply:
Doesn't seem like it. The generated videos seem to have yuv420p either way.